Improve Code Quality and Comment Clarity

FLASH2332 opened the pull request:
I have made a few changes as mentioned in #1, but I couldn't find certain parts of the code as they seem to have been updated in the latest version. How should I handle this?

FLASH2332 committed:
Correct import ordering for better readability.

FLASH2332 committed:
Use backticks for raw strings in Println.

FLASH2332 committed:
provide descriptive code comments.

Ashrockzzz2003 reviewed (commented):
Hey @FLASH2332 
Try to cover as many as you can and mark those as resolved in #1. I saw your comments there and it felt legit too.
You're free to make more improvements to the codebase too!

Let us know whn you're done and we'll merge this one. So far looks good.

FLASH2332 commented:
> Hey @FLASH2332 Try to cover as many as you can and mark those as resolved in #1. I saw your comments there and it felt legit too. You're free to make more improvements to the codebase too!
> 
> Let us know whn you're done and we'll merge this one. So far looks good.

Sure will look into it. Also I was not able to see option to mark conversation as resolved #1 .

FLASH2332 committed:
Provide descriptive code comments.

FLASH2332 commented:
![Screenshot 2024-12-29 202417](https://github.com/user-attachments/assets/3e635ff3-511a-41cd-9742-6687614392b7)
I found few changes that can be made to the error string. But I want to confirm if they are left in this format for some purpose?

Ashrockzzz2003 commented:
> ![Screenshot 2024-12-29 202417](https://private-user-images.githubusercontent.com/148314235/399169025-3e635ff3-511a-41cd-9742-6687614392b7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU0ODQ1NzcsIm5iZiI6MTczNTQ4NDI3NywicGF0aCI6Ii8xNDgzMTQyMzUvMzk5MTY5MDI1LTNlNjM1ZmYzLTUxMWEtNDFjZC05NzQyLTY2ODc2MTQzOTJiNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIyOVQxNDU3NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZWY2ZWI1NmZjMDAxYTVhNjQ2ZGRkOGI0YWMxYjBmY2M3YWUyZWUxMjdlOGEyZTNlNzVkZWUzNGJmMjQ1ZmQ5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Fvqvo68F1UtWl33z8uXmmvBIWBMUhw93X6ULWMBF7z8) I found few changes that can be made to the error string. But I want to confirm if they are left in this format for some purpose?

@Abhinav-ark look pls

FLASH2332 commented:
> > ![Screenshot 2024-12-29 202417](https://private-user-images.githubusercontent.com/148314235/399169025-3e635ff3-511a-41cd-9742-6687614392b7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU0ODQ1NzcsIm5iZiI6MTczNTQ4NDI3NywicGF0aCI6Ii8xNDgzMTQyMzUvMzk5MTY5MDI1LTNlNjM1ZmYzLTUxMWEtNDFjZC05NzQyLTY2ODc2MTQzOTJiNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIyOVQxNDU3NTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZWY2ZWI1NmZjMDAxYTVhNjQ2ZGRkOGI0YWMxYjBmY2M3YWUyZWUxMjdlOGEyZTNlNzVkZWUzNGJmMjQ1ZmQ5JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Fvqvo68F1UtWl33z8uXmmvBIWBMUhw93X6ULWMBF7z8) I found few changes that can be made to the error string. But I want to confirm if they are left in this format for some purpose?
> 
> @Abhinav-ark look pls

Could you please confirm if I can proceed with the changes, @Abhinav-ark? I've reviewed the codebase and everything appears to be in order. If any changes are required, please let me know.

Abhinav-ark commented:
> ![Screenshot 2024-12-29 202417](https://private-user-images.githubusercontent.com/148314235/399169025-3e635ff3-511a-41cd-9742-6687614392b7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU1NzQxMDQsIm5iZiI6MTczNTU3MzgwNCwicGF0aCI6Ii8xNDgzMTQyMzUvMzk5MTY5MDI1LTNlNjM1ZmYzLTUxMWEtNDFjZC05NzQyLTY2ODc2MTQzOTJiNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjMwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIzMFQxNTUwMDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NmZjOGRjMWQyODlhNmIzOTQwNjY5OTU0MGEyMDgyMTI5MzI0NWFkOTkyYTVmNzNkYmUyZjQ3ZmY4MTc4NjE2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.a0PesoW7CgYVo4mB0pBDVat7eqM9p6CSZXIMhyrsxkc) I found few changes that can be made to the error string. But I want to confirm if they are left in this format for some purpose?

No reason for this format @FLASH2332.

FLASH2332 commented:
Ok will change them Thanks!

FLASH2332 committed:
- Fix non-constant format string in error handling
- Fix error handling: format error messages to begin with a lowercase letter and end without punctuation as per Go guidelines

FLASH2332 commented:
- "Fix non-constant format string in error handling"
- "Fix error handling: format error messages to begin with a lowercase letter and end without punctuation as per Go guidelines"
- "Fix result of fmt.Errorf call not used"
Currently these are the fixes done in code base.

FLASH2332 commented:
> Hey @FLASH2332 Try to cover as many as you can and mark those as resolved in #1. I saw your comments there and it felt legit too. You're free to make more improvements to the codebase too!
> 
> Let us know whn you're done and we'll merge this one. So far looks good.

I have made some improvements to the codebase. Additionally, I explored this pull request (https://github.com/CSE-25/amrita_pyq/pull/15#issue-2662238678) and implemented a part of it. I believe there are no further improvements needed.

Let me know if you need any more adjustments!

Ashrockzzz2003 reviewed (changes requested):
Overall looks good. 
One last thing
Try running `gofmt -w cmd/`

FLASH2332 committed:
Format code in cmd package using gofmt

FLASH2332 commented:
Done

Ashrockzzz2003 reviewed (approved):
LGTM

FLASH2332 commented:
@Abhinav-ark is there anything else I should change ?

Ashrockzzz2003 commented:
/bounty 30 @FLASH2332

Ashrockzzz2003 commented:
> @Abhinav-ark is there anything else I should change ?

He's not available for review. Merging this as everything looks good. Thanks for the contribution @FLASH2332
